runtime/src/runtime/mod.rs

```diff
@@ -155,7 +155,7 @@ pub trait Runtime: Primitives + Verifier + RuntimePolicy {
     fn get_state_root(&self) -> Result<Cid, ActorError>;
 
     /// Sets the state-root.
-    fn set_state_root(&mut self, root: &Cid) -> Result<(), ActorError>;
+    fn set_state_root(&self, root: &Cid) -> Result<(), ActorError>;
```

Comment:
What are your thoughts about the right pattern here, for this and other items? I see two options of:
- the `Runtime` generally presents immutable `self`, as an abstraction over static syscalls, and mocks etc must use `Cell`s etc to get interior mutability
- the `Runtime` generally presents mutable `self` so that non-static implementations are easy to create directly.

Reply:
In general, I'd go with the former as it's strictly more flexible.